### ISSUE TITLE
Make sure `recommended.js` is up-to-date and dependecies are up-to-date

### ISSUE THREAD

**Issue by khiga8** [primer/eslint-plugin-primer-react#56]
The component to tag mappings in [recommended.js](https://github.com/primer/eslint-plugin-primer-react/blob/main/src/configs/recommended.js#L23) have not been updated for a while, and is overdue for a review.

In addition, this library is behind on `jsx-a11y` and `eslint-plugin-github` dependency updates.

We have a mapping that is used in `eslint-plugin-jsx-a11y` and a mapping that is used in `eslint-plugin-github`. The latter should be deprecated since it was an experimental feature that is deprecated in the latest `eslint-plugin-github`.

* Are there any component to tag mappings that should be newly added? (Notably, `jsx-a11y` supports 1:1 component to tag mapping). However, we should be wary of adding some mappings that could raise a false positive. For example, if we map `Avatar` to `img`, but it already sets `alt` internally default, the ESLint rule may incorrectly flag that there's a missing `alt`.
* Are there any component to tag mappings that should be removed? 

## Acceptance criteria

* [ ] Bump eslint-plugin-github and eslint-plugin-jsx-a11y dependency to the latest
* [ ] Ensure that the configs are up-to-date.

**Comment by TylerJDev** [comment]
Hey @khiga8, this sounds good! 

Just to clarify, is the idea to remove the mappings under "github" in the [recommended.js file](https://github.com/primer/eslint-plugin-primer-react/blob/main/src/configs/recommended.js#L17C9-L17C9)?

As far as the mapping in `jsx-a11y`, we definitely could add additional mappings. Nothing comes to mind right now, but I can review our current list of components to see if any of them would benefit from being mapped.

**Comment by khiga8** [comment]
Hey @tylerjdev! We're going to release a new version of `eslint-plugin-github` hopefully tomorrow, once these two PRs are merged in:

* https://github.com/github/eslint-plugin-github/pull/448
* https://github.com/github/eslint-plugin-github/pull/454

So I would encourage bumping to that version once it's out!

Here's what changed:
* The existing mapping currently under `github` can be removed. that format has been deprecated thanks to Kendall's work in https://github.com/github/eslint-plugin-github/pull/449.
* If we want to define component mapping for accessibility rules now in `eslint-plugin-github` too, we now support the same format as `jsx-a11y`. They are documented in https://github.com/github/eslint-plugin-github#component-mapping-experimental.

**Comment by TylerJDev** [comment]
Sounds good! I'll create a draft PR that should include some of the changes suggested!

**Comment by TylerJDev** [comment]
Closing per #67.

**Comment by khiga8** [comment]
@TylerJDev should we keep this open until the false positives flags are fixed? 🤔

**Comment by khiga8** [comment]
Ooh that sounds good! Just want to make sure your follow-up efforts are tracked! :)
